docs_src/design/adr/core/0022-UoM.md

```diff
@@ -0,0 +1,199 @@
+
+### Specifying the Units of Measure
+
+Units of measure allowed by the instance of EdgeX will be specified in a configuration file (in TOML format).  This can either be specified directly in the core metadata configuration file (but then hard to change when using a containerized or snap form of the application) or in a separate and external configuration TOML file (making it easier for an adopter to provide units of measure even when the application is containerized or snapped up).  
+
+The units of measure in the configuration file can be attributed, optionally, to a specification, document, or other UoM definition source.  The `source` only helps provide the location of documentation about the origins and details of the units specified for the reader, but it will not be used or checked by EdgeX.  An optional default source can be provided at the top level configuration (as shown in the examples below) so that other sources are only need when there specific units used that are not found in the default source.
+
+The units of measure can be categorized for better organization and to allow for different sources to be specified for different units.  The categories are defined by the TOML section names (the UoM dot labels).
+
+**Sample TOML unit of measure configuration**
+
+``` toml
+[Uom]
+    source="reference to source for all UoM if not specified below"
+    [Uom.temperature]
+    source="www.weather.com"
+    values="C,F,K"
+    [Uom.weights]
+    source="www.usa.gov/federal-agencies/weights-and-measures-division"
+    values="lbs,ounces,kilos,grams"
+```
+
+!!! Note
+        In the future, the source may be used by EdgeX to dynamically pull in units of measure rather than explicitly listing them in a configuration file.  As there is no standard for this type of data exchange today, this goes beyond the complexity of EdgeX needs, and would be left to the adopter to implement this at their discretion if desired.
+
+#### External UoM definition
+
+When the units of measure are specified in a separate (i.e. external definition) file, the location of the UoM file (specified by URI) will be specified in core metadata's configuration (currently in `res/configuration.toml`) - see example A below.  
+
+**Example A Metadata Configuration - external definition of UoM**
+``` toml
+[Writable]
+    [Writable.UoM]
+    validation=off  ## off by default
+
+## in the non-writable area - example external file specified to units of measure
+[UoM]
+    UoMFile="file::/tmp/edgex/metadata/uom.toml"  # the UoMFile location is specified as a URI.
+```
+
+The location of the external UoM file (specified by URI) should point to an accessible website, shared volume, volume mount, etc. in order to allow for the adopter to provide the units of measure independently during configuration/setup of the EdgeX instance without requiring a build of the metadata service or a reconstruction of the Docker image/container.
```

Comment:
As mentioned on another comment and by Tony, a reconstruction of Docker images may not be needed.